tests/js/set-tests.js

```diff
@@ -0,0 +1,392 @@
+            realm.create(people_schema.name, {Persons: [
+                {FirstName: "Joe",  Age: 4},
+                {FirstName: "Sue",  Age: 53},
+            ]});
+        });
+
+        let people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(2, people[0].Persons.length, "Persons Set length should be 2");
+
+        // add another person
+        realm.write(() => {
+            people[0].Persons.add({FirstName: 'bob', Age: 99});
+        });
+
+        people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(3, people[0].Persons.length, "Persons Set length should be 3");
+//        console.log("Joe's name:  " + people.Persons[0].FirstName);
+        let onfe = people[0].Persons.get(0);
+
+        let filteredsues = people[0].Persons.filtered('FirstName = "Sue"');
+
+        console.log("Sue's age:  " + filteredsues[0].Age);
+
+
+        // TODO:  The tests below are waiting for LinkedObj support in Mixed
+
+//         let one = people[0].Persons.get(0);
+//         let oij = people[0].Persons.has({FirstName: "Sue",  Age: 53});
+
+//         realm.write(() => {
+//             people = realm.objects(people_schema.name);
+//             people[0].Persons.delete(one);
+// //            people[0].Persons.delete({FirstName: "Sue",  Age: 53});
+//         });
+//         people = realm.objects(people_schema.name);
+
+//         TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+//         TestCase.assertEqual(2, people[0].Persons.length, "Persons Set length should be 2");
+
+        // TODO: add another 'People'
+    },
+
+    async testSetUtilityFunctions() {
+        const people_schema = {
+            name: "Person",
+            properties: {
+                Names: "string<>"
+            }
+        };
+
+        const realm = new Realm({ schema: [people_schema] });
+        const schema = realm.schema;
+
+        //
+        // Set.has() functionality
+        //
+
+        realm.write(() => {
+            // put some names in our database
+            realm.create(people_schema.name, {
+                Names: ["Kenneth", "Andrew", "Steffen"]
+            });
+        });
+
+        let teams = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, teams.length, "There should be only one JS team");
+        let js_team = teams[0];
+        TestCase.assertEqual(3, js_team.Names.length, "There should be 3 people in the JS team");
+        TestCase.assertEqual(true, js_team.Names.has("Kenneth"), "Kenneth should be in the JS team");
+        TestCase.assertEqual(false, js_team.Names.has("Jason"), "Jason shouldn't be in the JS team");
+
+        // add one JS team member, delete another one
+        realm.write(() => {
+            js_team.Names.add("Kræn").delete("Kenneth");
```

Comment:
I don't think he would like that 😆 

In all seriousness, Alice, Bob and Charlie are often better used here.

Reply:
Awww :)